src/arch/x86.rs

```diff
@@ -98,7 +140,8 @@ pub unsafe fn init(stack: &Stack, f: unsafe extern "C" fn(usize) -> !) -> StackP
   let mut sp = StackPointer(stack.base() as *mut usize);
   push(&mut sp, 0xdead0cfa); // CFA slot
   push(&mut sp, f as usize); // function
-  push(&mut sp, trampoline_1 as usize);
+  push(&mut sp, trampoline_1 as usize + 2); // skip the 2 nops
```

Comment:
Please add a comment here explaining that we set up the stack as-if the first trampoline has already execited.